elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,1089 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+ * @param array $query
+ *   The query.
+ *
+ * @return SimpleXMLElement|FALSE
+ *   The response from the REST API as a SimpleXMLElement, or FALSE if request
+ *   failed.
+ */
+function _xml_form_elements_creative_commons_query_api($path, $query = array()) {
+  $response = drupal_http_request(url(XML_FORM_ELEMENTS_CREATIVE_COMMONS_API_URL . $path, array('query' => $query)));
+  if ($response->code != 200) {
+    watchdog('xml_form_elements', 'The Creative Commons API responded with code %code.', array('%code' => $response->code), WATCHDOG_WARNING);
+    return FALSE;
+  }
+  return simplexml_load_string($response->data, 'SimpleXMLElement');
+}
+
+/**
+ * Ask the creative commons REST API for a 'by' license.
+ *
+ * Left as a function for use by community modules.
+ *
+ * @deprecated As of release 7.x-1.11, this function remains only for backwards
+ *   compatibilty. New code should use
+ *   xml_form_elements_creative_commons_query_api_get() instead.
+ *   Update your code, as this function might be removed.
+ *
+ * @param string $commercial
+ *   'y' to allow, 'n' to disallow.
+ * @param string $derivatives
+ *   'y' to allow, 'n' to disallow, 'sa' to share alike.
+ * @param string $jurisdiction
+ *   Legal jurisdiction code.
+ * @param bool $disabled
+ *   If TRUE, return FALSE; otherwise perform query.
+ *
+ * @return mixed
+ *   SimpleXMLElement the return from the REST API.
+ *   FALSE if failed request.
+ */
+function xml_form_elements_get_creative_commons($commercial, $derivatives, $jurisdiction, $disabled = FALSE) {
+  // Show deprecated message.
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  $deprecated_message = islandora_deprecated('7.x-1.11', t('Use @newfunction() instead.', array('@newfunction' => 'xml_form_elements_creative_commons_query_api_get')));
+  trigger_error(filter_xss($deprecated_message), E_USER_DEPRECATED);
+  // Return FALSE if disabled.
+  if ($disabled) {
+    return FALSE;
+  }
+  // This function only pertains to 'by' licenses.
+  $api_path = 'license/standard/get';
+  // Prepare query options.
+  $api_options = array(
+    'commercial' => $commercial,
+    'derivatives' => $derivatives,
+    'jurisdiction' => $jurisdiction,
+  );
+  // Return API response.
+  return xml_form_elements_creative_commons_query_api_get($api_path, $api_options);
+}
+
+/**
+ * Gets the 'by' license URI for the element.
+ *
+ * @deprecated As of release 7.x-1.11, this function remains only for backwards
+ *   compatibilty. New code should use
+ *   xml_form_elements_creative_commons_get_uri_from_values() instead.
+ *   Update your code, as this function might be removed.
+ *
+ * @param string $commercial
+ *   'y' to allow, 'n' to disallow.
+ * @param string $derivatives
+ *   'y' to allow, 'n' to disallow, 'sa' to share alike.
+ * @param string $jurisdiction
+ *   Legal jurisdiction code.
+ * @param bool $disabled
+ *   If TRUE, return empty string; otherwise generate the URI.
+ *
+ * @return string
```

Comment:
or empty if disabled?

Reply:
@DiegoPino again, this is deprecated function henceforth. I figured the empty string `''` is still a `string` regarding type...

Reply:
The value of the element == '' ?

Reply:
If you are ok with that i'm ok

Reply:
@DiegoPino not sure I got your issue, then...

Reply:
Just being picky, if disabled the function is not "really" returning the value of the element (line 1197)... forget it.